### PR TITLE
Change common tags to use Set instead of Add method. Add Span Kind.

### PR DESCRIPTION
**Pull request by yurishkuro**
Brings Go in parity with Python, where span kind tags have been defined for a while.

### CODE REVIEW

File: ext/tags.go [hunk 1]
```diff
@@ -23,25 +23,54 @@ var (
...
+
+	// SpanKindRPCServer marks a span representing the server-side of an RPC
+	// or other remote call
+	SpanKindRPCServer = SpanKindEnum("s")
 )
 
+// ---
+
+// SpanKindEnum represents common span types
+type SpanKindEnum string
```
Comment (tschottdorf) [inline ext/tags.go]: The plain types here make this pretty confusing to use - why not something like

```go
type SpanKind interface {
  spanKindMarker()
  Set(span opentracing.Span, value string)
}

type spanKindImpl struct {
  name string
}
func (sk *spanKindImpl) spanKindMarker() {}
func (sk *...) Set(...) { ... }

var SpanKindRPCClient SpanKind = (&spanKindImpl{name:"s"})(nil)
```
func (

Comment (yurishkuro) [inline ext/tags.go]: I've added new values using the same pattern used by the other ones. That pattern has been agreed upon back in Dec https://github.com/opentracing/opentracing-go/commit/3e2f0d7c16a7aa88576ef114a638cbaf3471f8c9#diff-0c56d0f1cdc8e3b8ba3be800367de5ee

If you think it's worth changing, e.g. from performance considerations, I think it should be a different PR.

**Comment by bensigelman** [comment]
@yurishkuro forgive me for being lazy and not just checking myself, but is this PR compliant with @dkuebric's pending https://github.com/opentracing/opentracing.github.io/pull/61 ?

**Comment by yurishkuro** [comment]
@bensigelman yes, it is (by the nature of just following what's already done in Python). This is form Dan's pr:

```
* `span.kind` - string
    - One of `c` or `s`, indicating if this span represents a client or server
```

**Comment by bensigelman** [comment]
Well, this LGTM once you and @tschottdorf are on the same page...

File: ext/tags.go (hunk 1 above)
Comment (tschottdorf) [inline ext/tags.go]: SGTM
